patch/patcher.go

```diff
@@ -276,14 +280,24 @@ func (p *Patcher) scanField(f *protogen.Field) {
 		}
 		newName = lint.Name(newName, lints.InitialismsMap())
 	}
+	// check embed for message types
+	embed := f.Message != nil && f.Oneof == nil && opts.GetEmbed()
+	if embed {
```

Comment:
This should be moved above line 277 where it attempts to lint the `newName`. Please make this change and add a test case for linting an embedded field.